### PR TITLE
Aashish/fixes

### LINKED ISSUE #60: Open playground from courses page
1. Directly open the playground from the courses page with the code, when someone click on python code
2. If no code exists in playground, directly open
3. If code exists in playground, show a popup:

Are you sure you want to replace the existing code?
Yes (new code)
No, (existing code)

### PR DESCRIPTION
**Pull request by aashish-aadarsh**
PR for issues fixes #60  #75  #72  #58 #56

### CODE REVIEW

File: app/src/main/java/org/merakilearn/ui/home/HomeViewModel.kt [hunk 1]
```diff
@@ -14,6 +14,10 @@ class HomeViewModel(private val applicationRepo:ApplicationRepo) : ViewModel() {
         emit(applicationRepo.fetchOtherCourseData())
     }
 
+    fun fetchUpcomingClass(langCode: String?) = liveData {
```
Comment (suhaibroomy) [inline app/src/main/java/org/merakilearn/ui/home/HomeViewModel.kt:17]: Why is this needed here?

File: app/src/main/java/org/merakilearn/util/AppUtils.kt [hunk 2]
```diff
@@ -153,5 +157,19 @@ object AppUtils {
         """.trimIndent()
     }
 
+    fun saveLanguageConfig(data: String, context: Context) {
```
Comment (suhaibroomy) [inline app/src/main/java/org/merakilearn/util/AppUtils.kt:160]: Both these methods can be removed in favour of ConfigRepo

File: learn/src/main/res/values/dimens.xml [hunk 3]
```diff
@@ -29,4 +29,5 @@
     <dimen name="dimen_100_dp">100dp</dimen>
     <dimen name="dimen_slider_layout">100dp</dimen>
     <dimen name="dimen_youtube_view_height">250dp</dimen>
+    <dimen name="dimen_80_dp">80dp</dimen>
```
Comment (suhaibroomy) [inline learn/src/main/res/values/dimens.xml:32]: creating dimen with value as its name isn't useful, its' the same as using hardcoded values. Please create dimens with meaningful name and if they can not be reused anywhere then they should be hardcoded

**Review by suhaibroomy — CHANGES_REQUESTED** [review]
Please create different PR for different issues. Its difficult to understand which change corresponds to which issue

File: app/src/main/java/org/merakilearn/ui/home/HomeViewModel.kt (hunk 1 above)
Comment (aashish-aadarsh) [reply app/src/main/java/org/merakilearn/ui/home/HomeViewModel.kt:17]: For langugage filter

Comment (suhaibroomy) [reply app/src/main/java/org/merakilearn/ui/home/HomeViewModel.kt:17]: Yes but this should be handled by DiscoverViewModel, not HomeViewModel @aashish-aadarsh

**Review by suhaibroomy — APPROVED** [review]

**Pull request merged**